File: client-app/src/models/UserProfile.ts
import { UserFollowing } from './Follower';
import { Photo } from './Photo';

export interface UserAbout {
  displayName: string;
  bio?: string;
}

export interface UserProfile extends UserAbout {
  id: string;
  email: string;
  userName: string;
  image?: string;
  photos: Photo[];
  followers: UserFollowing[];
  followings: UserFollowing[];
}

File: Domain/Photos/UserProfile.cs
namespace Domain.Photos;

public class UserProfile
{
    public string Id { get; set; }
    public string Email { get; set; }
    public string UserName { get; set; }
    public string DisplayName { get; set; }
    public string Bio { get; set; }
    public IList<Photo> Photos { get; set; }
}

public class UserProfileFullInfo : UserProfile
{
    public string Image { get; set; }
}